src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml

```diff
@@ -550,3 +550,11 @@ services:
       - '@prestashop.core.hook.dispatcher'
       - '@prestashop.core.grid.query.doctrine_query_parser'
       - 'authorized_application'
+
+  prestashop.core.grid.data.factory.customer_groups:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Data\Factory\DoctrineGridDataFactory'
+    arguments:
+      - '@prestashop.core.grid.query_builder.customer_groups' # service id of our query builder
```

Comment:
And and I forgot but we also have some yamllint that you can run locally with `yamllint -c .github/workflows/yamllint/.yamllint.yml src`

You'll have to install `yamllint` locally